### PR TITLE
Thermite tweak

### PR DESCRIPTION
**Pull request by HMBGERDO**
## What Does This PR Do

This PR changes thermite melting logic. Now melting time and melting progress is on thermite volume. Now it uses 100 units to melt r wall and 20 units to melt default wall. Melting r wall is 20 seconds, melting basic wall is 4 seconds.
When you start melting, it doesnt remove the wall, instead it is damaging it every second, making welder sound.

This does not affect T4, it still melts wall in 3 seconds(or whatever its time is)

## Why It's Good For The Game

I really dont like the idea of 3chem reagent being able to melt everything in one drop. This PR forces you to at least get MORE thermite to melt r walls, making raiding shit like AI sat harder. 

## Testing

Compiled, thermitted

## Changelog
:cl:
tweak: Thermite now requires volume to melt walls. 20 units to basic metal wall, 100 units to reinforced wall. Its melting time now is also based on wall type and its health. Every second thermite applies 30 damage to a wall and uses 5u of it. Thermite now is also making welding sounds every second instead of melting wall silently.
/:cl:

### CODE REVIEW

**Comment by 1080pCat** [comment]
How does this effect the phlog flash fire interaction, if at all?
(splash termite then phlog onto a wall melting it considerably faster than normal termite)

**Comment by HMBGERDO** [comment]
> How does this effect the phlog flash fire interaction, if at all? (splash termite then phlog onto a wall melting it considerably faster than normal termite)

this PR doesnt affect fireflash_sm proc if you're talking about that(instantly melting walls in blue fire) so means it will do the same

**Review by matttheficus — APPROVED** [review]

File: code/game/turfs/simulated/walls.dm [hunk 1]
```diff
@@ -261,16 +261,40 @@
...
+			var/turf/simulated/floor/F = src
+			F.burn_tile()
+			F.icon_state = "plating"
+			break
+		take_damage(30)
+		playsound(src, 'sound/items/welder.ogg', 100, TRUE)
+		sleep(1 SECONDS)
+	if(iswallturf(src))
+		melting = FALSE
+	if(O)	qdel(O)
```
Comment (S34NW) [inline code/game/turfs/simulated/walls.dm:297]:
```suggestion
	if(O)
		qdel(O)
```

**Review by S34NW — CHANGES_REQUESTED** [review]

**Comment by YoureNotTheSharpest** [comment]
Wait this wasnt already merged?

File: code/game/turfs/simulated/walls.dm [hunk 2]
```diff
@@ -18,6 +18,7 @@
 	var/damage_cap = 100 //Wall will break down to girders if damage reaches this point
 
 	var/global/damage_overlays[8]
+	var/melting = FALSE
```
Comment (lewcc) [inline code/game/turfs/simulated/walls.dm:21]: Please autodoc this

File: code/game/turfs/simulated/walls.dm [hunk 3]
```diff
@@ -261,16 +261,41 @@
...
+		var/turf/simulated/floor/F = src
+		F.burn_tile()
+		F.icon_state = "plating"
 		if(O)	qdel(O)
+		return
+
+	melting = TRUE
+	while(TRUE)
+		if(!reagents.get_reagent_amount("thermite"))
+			break
```
Comment (lewcc) [inline code/game/turfs/simulated/walls.dm:283]:
```suggestion
	while(reagents.get_reagent_amount("thermite") > 0)
```

**Pull request merged**